### PR TITLE
DSG-1635/Upgrade-to-Spring-Boot-3.1

### PR DESCRIPTION
**Pull request by afcalderonm**
## Description

Upgrade SpringBoot version

[___JIRA_TICKET_TITLE___](//jira_url)

## Types of changes

Patch

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [x] Refactor or cleanup (functionality unchanged)
  Minor
- [ ] New feature (non-breaking change which adds functionality)
  Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [x] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

**Review by asultan001 — APPROVED** [review]
LGTM but I'd like for @deibys-nuvalence to give it another look just in case.

File: service/build.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 plugins {
-	id 'org.springframework.boot' version '2.7.8'
-	id 'io.spring.dependency-management' version '1.1.0'
+	id 'org.springframework.boot' version '3.1.3'
```
Comment (deibys-nuvalence) [inline service/build.gradle:2]: At the time , I upgrade to,  3.1.3 was the latest , but now 3.1.4 is the newest one . I dont think there would be any issue upgrading to 3.1.4

File: service/build.gradle [hunk 2]
```diff
@@ -35,13 +35,13 @@ dependencies {
...
 	implementation 'org.springframework.boot:spring-boot-starter-security'
-	implementation 'com.google.cloud:spring-cloud-gcp-starter-secretmanager:3.4.4'
-	implementation 'io.nuvalence.dsgov:dsgov-auth:0.1.10'
-	implementation 'io.nuvalence.dsgov:dsgov-logging:0.0.11'
+	implementation 'com.google.cloud:spring-cloud-gcp-starter-secretmanager:4.7.2'
+	implementation 'io.nuvalence.dsgov:dsgov-auth:1.0.0'
+	implementation 'io.nuvalence.dsgov:dsgov-logging:1.0.0'
 	implementation 'com.google.cloud:google-cloud-logging-logback:0.130.7-alpha'
 
 	// version bump to address known CVEs
```
Comment (deibys-nuvalence) [inline service/build.gradle:43]: This dependency was added to override the one pulled by dependency setup (which at the time was a lower version)
implementation 'org.yaml:snakeyaml:2.2'
With spring boot 3 upgrade, this library with this versions or higher should be brought up
I guess if you check the dependency tree and that libraries is already pulled with this version, you can remove these direct dependency inclusion

**Review by deibys-nuvalence — CHANGES_REQUESTED** [review]
Just move to 3.1.4
Also be aware when you deploy to dev, the artifact repository should be updated. At least I had problems when deploying to dev
 repositories {
        mavenCentral()
        maven {
            url 'https://oss.sonatype.org/content/repositories/snapshots/'
        }
    }
as this currently brings snapshot dependencies

**Review by deibys-nuvalence — APPROVED** [review]

File: service/build.gradle (hunk 1 above)
Comment (afcalderonm) [reply service/build.gradle]: Done

File: service/build.gradle (hunk 2 above)
Comment (afcalderonm) [reply service/build.gradle:43]: i tried removing and check in the dependency tree and there was brought up with the springboot dependency but it still using the 1.33 version that has a vulnerability  so i think is better to keep this one

**Review by deibys-nuvalence — APPROVED** [review]

**Pull request merged**